AmplifyPlugins/API/Sources/AWSAPIPlugin/Configuration/AWSAPICategoryPluginConfiguration.swift

```diff
@@ -160,6 +180,25 @@ public struct AWSAPICategoryPluginConfiguration {
         return endpoints
     }
 
+    private static func endpointsFromConfig(
+        config: AmplifyOutputsData,
+        apiAuthProviderFactory: APIAuthProviderFactory,
+        authService: AWSAuthServiceBehavior
+    ) throws -> [APIEndpointName: EndpointConfig] {
+        var endpoints = [APIEndpointName: EndpointConfig]()
+
+        if let data = config.data {
+            let name = "dataCategory"
+            let endpointConfig = try EndpointConfig(name: name,
+                                                    config: data,
+                                                    apiAuthProviderFactory: apiAuthProviderFactory,
+                                                    authService: authService)
+            endpoints[name] = endpointConfig
+        }
+
+        return endpoints
```

Comment:
great question. A lot of the other categories will throw a `PluginError.configuration` error when it cannot find the category to configure with. Configuring **AWSAPIPlugin** should be the same, so instead of returning an empty endpoints, it should throw PluginError. I'll update the code

In the future, **AWSAPIPlugin** will support not just the endpoint set up from the data category but endpoints from other categories, such as `API` for REST/API Gateway endpoints. Our logic should then map to endpoints from these two places, `data` and `api` category, and if at the end of it `endpoints` is still empty then we should throw the PluginError.

Reply:
Updated